### PR TITLE
Role: add StateBasedAction

### LINKED ISSUE #3685: WOE: Role unattach
just to be clear, i will handle the unattach of the Roles later when we get more clarification on what happens when multiple roles are attached at the same time.

if they are attached + unattached in Sequence
or if they are attached at once and the others unattached at once
or if they are just unattached as SBA

_Originally posted by @Hanmac in https://github.com/Card-Forge/forge/issues/3636#issuecomment-1691219344_
            
for this we need answers from Rules Manager and/or CR update

### PR DESCRIPTION
**Pull request by Hanmac**
Closes #3685 

the Send to Graveyard should only be needed as StateBasedAction

as per rules, it looks like they are directly send to GY instead of getting unattached first? That might matter?
I still unattach them first for better trigger

As for the Cards that makes the Roles, @Agetian AI might need some logic to not add Roles to cards that already has Roles? (except the Cursed one?)

### CODE REVIEW

**Comment by tool4ever** [comment]
should be good besides AI

**Comment by Hanmac** [comment]
@Agetian can that be merged yet?

i think the general AI about Role Auras can be polished later

**Comment by tool4ever** [comment]
I've added a initial part so at least it doesn't cheat with stuff like _Spiteful Hexmage_

File: forge-ai/src/main/java/forge/ai/ability/TokenAi.java [hunk 1]
```diff
@@ -359,4 +372,42 @@ public static Card spawnToken(Player ai, SpellAbility sa) {
         return result;
     }
 
+    private boolean tgtRoleAura(final Player ai, final SpellAbility sa, final Card tok, final boolean mandatory) {
+        boolean isCurse = tok.getFirstAttachSpell().getParamOrDefault("AILogic", "").equals("Curse");
```
Comment (Hanmac) [inline forge-ai/src/main/java/forge/ai/ability/TokenAi.java:376]: The SpellAbility that creates the token also should have AILogic Curse

imo that should be checked first

**Comment by Hanmac** [comment]
i think that AI is good for now

about the FristAttachSpell i will probably touch that to a later day when changing EnchantKeyword

**Pull request merged**